### ISSUE TITLE
The method 'detach' isn't defined for the class 'FocusScopeNode'

### ISSUE THREAD

**Issue by fvisticot** [AndreHaueisen/flushbar#44]
[✓] Flutter (Channel master, v1.5.5-pre.15, on Mac OS X 10.14.4 18E226, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.4)
[✓] VS Code (version 1.31.1)

I get this compilation error:
- 'FocusScopeNode' is from 'package:flutter/src/widgets/focus_manager.dart' ('file:///Users/fsdf/Tools/flutter/packages/flutter/lib/src/widgets/focus_manager.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'detach'.
        focusNode.detach();
                  ^^^^^^

**Comment by AndreHaueisen** [comment]
Hi @fvisticot 
You are on the master channel. This is a breaking change so I can't do anything until it hits stable otherwise it will break the package for everyone else.
Thanks for reporting though. I won't be caught with my guard down.

**Comment by rahulmanthena** [comment]
Do you have any workaround until then :)

**Comment by AndreHaueisen** [comment]
Sorry man. I don't.
Is it absolutely necessary for you to stay on the master channel?
I suggest the beta.

**Comment by zachstarnes** [comment]
Hey @AndreHaueisen I am getting this error on version 1.5.0 installing with pub. Since you do not have a fix for this at the moment what version number should I use in my pubspec.yaml

**Comment by AndreHaueisen** [comment]
You should use flutter stable or beta version.

**Comment by vanelizarov** [comment]
@AndreHaueisen I think it won't break anything if we call `dispose()` method on `focusNode` instead of `detach()`. The [doc](https://master-api.flutter.dev/flutter/widgets/FocusNode-class.html) says:

> The attachment will automatically be detached in dispose().

**Comment by AndreHaueisen** [comment]
It needs `FocusScopeNode`. Not `FocusNode`.
`FocusScopeNode` does not have `dispose()` yet.

**Comment by vanelizarov** [comment]
@AndreHaueisen aah, yeah, I see, messed up a little bit 🙃

Have you seen [this](https://github.com/flutter/flutter/issues/31784#issuecomment-487678661) comment? Can it help?

**Comment by masewo** [comment]
Now it happens with flutter dev channel too. Not only master anymore.
Implemented [this](https://github.com/flutter/flutter/issues/31784#issuecomment-487678661) comment in my fork:
https://github.com/masewo/flushbar
Seems to be working so far.
Very likely that we do not need FocusAttachment here.
Maybe it should sufficient to replace detach by dispose.
Edit: Working without FocusAttachment too.

**Comment by AndreHaueisen** [comment]
@vanelizarov The comment helped me understand what has changed. Thank you very much.
@masewo I'm aware of the solution by now, but this does not work on beta or stable since we don't have `FocusAttachment` yet. I can't push the fix because it would break the package for those who are not in the dev or master.
I will wait for it to reach beta at least since the chance for another breaking change is lower.

For now, either migrate to beta or stable or use maseo's version.

**Comment by vanelizarov** [comment]
@AndreHaueisen maybe you can add this feature to e.g. “dev” version of the package by creating separate branch and let the users who have faced this issue install your lib from Github instead of Pub?

**Comment by AndreHaueisen** [comment]
@vanelizarov Agreed.

For those who don't want to change to stable or beta channels, install the package using:

```
flushbar:
    git:
      url: https://github.com/AndreHaueisen/flushbar
      ref: dev_version
```

**Comment by AndreHaueisen** [comment]
Thanks for the feedback guys 👍

**Comment by yohom** [comment]
Hi @AndreHaueisen, Flutter has hit its new beta(1.6.3), can you release a new version to fix this problem?

**Comment by AndreHaueisen** [comment]
The fix is live at version 1.6.0
Everyone that needs to stay in the stable channel should keep using version 1.5.3

**Comment by zinwalin** [comment]
> The fix is live at version 1.6.0
> Everyone that needs to stay in the stable channel should keep using version 1.5.3

thanks.
